### PR TITLE
Migrate to structured logging Part 2

### PR DESCRIPTION
**Pull request by arvidOtt**
What type of PR is this?
/kind feature

What this PR does / why we need it:
Structured Logging migration for
```
pkg/crossnodepreemption/cross_node_preemption.go:3
pkg/coscheduling/core/core.go:9
pkg/coscheduling/coscheduling.go:15
```

Which issue(s) this PR fixes:
#120

### CODE REVIEW

File: pkg/coscheduling/core/core.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ func NewPodGroupManager(pgClient pgclientset.Interface, snapshotSharedLister fra
 // 2. the total number of pods in the podgroup is less than the minimum number of pods
 // that is required to be scheduled.
 func (pgMgr *PodGroupManager) PreFilter(ctx context.Context, pod *corev1.Pod) error {
-	klog.V(5).Infof("Pre-filter %v", pod.Name)
+	klog.V(5).InfoS("Pre-filter", "pod", klog.KObj(pod))
 	pgFullName, pg := pgMgr.GetPodGroup(pod)
 	if pg == nil {
 		return nil
```
Comment (Huang-Wei) [inline pkg/coscheduling/core/core.go:104]: (not related to this line)

Seems you missed L108.

File: pkg/coscheduling/core/core.go [hunk 2]
```diff
@@ -140,7 +140,7 @@ func (pgMgr *PodGroupManager) PreFilter(ctx context.Context, pod *corev1.Pod) er
 	minResources[corev1.ResourcePods] = *podQuantity
 	err = CheckClusterResource(nodes, minResources, pgFullName)
 	if err != nil {
-		klog.Errorf("PreFilter pod group %v error: %v", pgFullName, err)
+		klog.ErrorS(err, "Failed to prefilter", "pod group", klog.KObj(pg))
```
Comment (Huang-Wei) [inline pkg/coscheduling/core/core.go:143]:
```suggestion
		klog.ErrorS(err, "Failed to PreFilter", "podGroup", klog.KObj(pg))
```

File: pkg/coscheduling/core/core.go [hunk 3]
```diff
@@ -191,16 +191,16 @@ func (pgMgr *PodGroupManager) PostBind(ctx context.Context, pod *corev1.Pod, nod
 	if pgCopy.Status.Phase != pg.Status.Phase {
 		pg, err := pgMgr.pgLister.PodGroups(pgCopy.Namespace).Get(pgCopy.Name)
 		if err != nil {
-			klog.Error(err)
+			klog.ErrorS(err, "pod group", klog.KObj(pgCopy))
```
Comment (Huang-Wei) [inline pkg/coscheduling/core/core.go:194]: The [guide](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#name-arguments) mentioned to use carmelCase name as the key:

```
Always use lowerCamelCase, for example use containerName and not container name or container_name.
```

So please use `podGroup` as the key (applies elsewhere).

File: pkg/coscheduling/core/core.go [hunk 4]
```diff
@@ -191,16 +191,16 @@ func (pgMgr *PodGroupManager) PostBind(ctx context.Context, pod *corev1.Pod, nod
...
 			return
 		}
 		patch, err := util.CreateMergePatch(pg, pgCopy)
 		if err != nil {
-			klog.Error(err)
+			klog.ErrorS(err, "pod group", klog.KObj(pg), "pod group", klog.KObj(pgCopy))
```
Comment (Huang-Wei) [inline pkg/coscheduling/core/core.go:199]: - the ErrorS has a 2nd parameter to log the "msg"
- logging one pod group is good enough as `klog.KObj` just dumps its "namespace/name".

```suggestion
			klog.ErrorS(err, "Failed to create merge patch", "podGroup", klog.KObj(pg))
```

File: pkg/coscheduling/core/core.go [hunk 5]
```diff
@@ -191,16 +191,16 @@ func (pgMgr *PodGroupManager) PostBind(ctx context.Context, pod *corev1.Pod, nod
...
 			return
 		}
 		if err := pgMgr.PatchPodGroup(pg.Name, pg.Namespace, patch); err != nil {
-			klog.Error(err)
+			klog.ErrorS(err, "pod group", klog.KObj(pg))
```
Comment (Huang-Wei) [inline pkg/coscheduling/core/core.go:203]:
```suggestion
			klog.ErrorS(err, "Failed to patch", "podGroup", klog.KObj(pg))
```

File: pkg/coscheduling/coscheduling.go [hunk 6]
```diff
@@ -77,7 +78,8 @@ func New(obj runtime.Object, handle framework.Handle) (framework.Plugin, error)
 
 	fieldSelector, err := fields.ParseSelector(",status.phase!=" + string(v1.PodSucceeded) + ",status.phase!=" + string(v1.PodFailed))
 	if err != nil {
-		klog.Fatalf("ParseSelector failed %+v", err)
+		klog.ErrorDepth(0,"ParseSelector failed" , err)
```
Comment (Huang-Wei) [inline pkg/coscheduling/coscheduling.go:81]: "Fatalf" should be replaced by "ErrorS and os.Exit()"

Comment (Huang-Wei) [reply pkg/coscheduling/coscheduling.go:81]: And also there is some linting issue here.

File: pkg/coscheduling/coscheduling.go [hunk 7]
```diff
@@ -161,31 +164,31 @@ func (cs *Coscheduling) PostFilter(ctx context.Context, state *framework.CycleSt
...
+		klog.V(4).InfoS("Pod does not belong to any group", "pod", klog.KObj(pod))
 		return &framework.PostFilterResult{}, framework.NewStatus(framework.Unschedulable, "can not find pod group")
 	}
 
 	// This indicates there are already enough Pods satisfying the PodGroup,
 	// so don't bother to reject the whole PodGroup.
 	assigned := cs.pgMgr.CalculateAssignedPods(pg.Name, pod.Namespace)
 	if assigned >= int(pg.Spec.MinMember) {
-		klog.V(4).Infof("%v pods of %v assigned", assigned, pgName)
+		klog.V(4).InfoS("Assigned pods", "pod group", klog.KObj(pg), "amount", assigned)
```
Comment (Huang-Wei) [inline pkg/coscheduling/coscheduling.go:175]:
```suggestion
		klog.V(4).InfoS("Assigned pods", "podGroup", klog.KObj(pg), "assigned", assigned)
```

**Comment by arvidOtt** [comment]
@Huang-Wei thanks for your feedback. I applied all changes requested 🙂

File: pkg/coscheduling/core/core.go [hunk 8]
```diff
@@ -98,14 +98,14 @@ func NewPodGroupManager(pgClient pgclientset.Interface, snapshotSharedLister fra
...
-	klog.V(5).Infof("Pre-filter %v", pod.Name)
+	klog.V(5).InfoS("Pre-filter", "pod", klog.KObj(pod))
 	pgFullName, pg := pgMgr.GetPodGroup(pod)
 	if pg == nil {
 		return nil
 	}
 	if _, ok := pgMgr.lastDeniedPG.Get(pgFullName); ok {
 		err := fmt.Errorf("pod with pgName: %v last failed in 3s, deny", pgFullName)
-		klog.V(6).Info(err)
+		klog.V(6).InfoS(err.Error(), "podGroup", pg)
```
Comment (Huang-Wei) [inline pkg/coscheduling/core/core.go:108]: It seems the caller also dumps the error, so let's eliminate the logging here and just return the error:

```
return fmt.Errorf("pod with pgName: %v last failed in 3s, deny", pgFullName)
```

File: pkg/coscheduling/coscheduling.go [hunk 9]
```diff
@@ -161,31 +164,31 @@ func (cs *Coscheduling) PostFilter(ctx context.Context, state *framework.CycleSt
...
+		klog.V(4).InfoS("Assigned pods", "podGroup", klog.KObj(pg), "assigned", assigned)
 		return &framework.PostFilterResult{}, framework.NewStatus(framework.Unschedulable)
 	}
 
 	// If the gap is less than/equal 10%, we may want to try subsequent Pods
 	// to see they can satisfy the PodGroup
 	notAssiginedPercentage := float32(int(pg.Spec.MinMember)-assigned) / float32(pg.Spec.MinMember)
 	if notAssiginedPercentage <= 0.1 {
-		klog.V(4).Infof("%.2f/100 pods of group %v have not assigned", float32(assigned*100), pgName)
+		klog.V(4).InfoS("Pods have not assigned to group", "podGroup", klog.KObj(pg), "percentage", notAssiginedPercentage)
```
Comment (Huang-Wei) [inline pkg/coscheduling/coscheduling.go:183]: nit:

```suggestion
		klog.V(4).InfoS("A small gap of pods to reach the quorum", "podGroup", klog.KObj(pg), "percentage", notAssiginedPercentage)
```

**Review by Huang-Wei — COMMENTED** [review]
Some nits. LGTM otherwise.

(you can squash the commits along with addressing the final nits)

**Comment by arvidOtt** [comment]
Done.

**Pull request merged**